workflow/saga.go

```diff
@@ -0,0 +1,62 @@
+package workflow
+
+type (
+    SagaOptions struct {
+        ParallelCompensation bool
+        ContinueWithError    bool
+    }
+
+    Saga interface {
+        AddCompensation(Context, interface{}, ...interface{}) Saga
+        Compensate() error
+    }
+
+    sagaImpl struct {
+        options         *SagaOptions
+        compensationOps []*compensationOp
+    }
+
+    compensationOp struct {
+        ctx      Context
+        activity interface{}
+        args     []interface{}
+    }
+)
+
+func NewSaga(options SagaOptions) Saga {
+    return &sagaImpl{options: &options}
+}
+
+func (s *sagaImpl) AddCompensation(ctx Context, activity interface{}, args ...interface{}) Saga {
```

Comment:
I don't think we want to store context here. For example if workflow is cancelled the contexts created from the root context will be cancelled and the compensation code is not going to execute correctly. I propose to add ctx to Compensate call and use context created from it using Workflow.newDisconnectedContext.